pubsub.go

```diff
@@ -621,6 +614,43 @@ func (p *PubSub) processLoop(ctx context.Context) {
 	}
 }
 
+func (p *PubSub) handlePendingPeers() {
+	select {
+	case <-p.newPeersSema:
+		defer func() {
+			p.newPeersSema <- struct{}{}
+		}()
+
+	default:
+		// contention, return and wait for the next notification without blocking the event loop
+		return
+	}
+
+	if len(p.newPeersPend) == 0 {
+		return
+	}
+
+	newPeers := p.newPeersPend
+	p.newPeersPend = make(map[peer.ID]struct{})
+
+	for pid := range newPeers {
```

Comment:
Are we concerned that if this map is large we could block the event loop?

Reply:
well, if it gets large enough to take a non trivial amount of time to process, we've got bigger problems! We are getting flooded with connections.